src/client/core/utils/dom.js

```diff
@@ -121,193 +309,66 @@ export function getElementDescription (el) {
-        withoutTabIndexArray = $elements.not($withTabIndex).toArray();
-
-    if ($iFrames.length)
-        return insertIFramesContentElements(withTabIndexArray, $iFrames).concat(insertIFramesContentElements(withoutTabIndexArray, $iFrames));
-
-    return withTabIndexArray.concat(withoutTabIndexArray);
-}
-
-function insertIFramesContentElements (elementsArray, $iFrames) {
-    var results         = [],
-        sortedIFrames   = sortElementsByTabIndex($iFrames),
-        iFramesElements = [];
-
-    for (var i = 0; i < sortedIFrames.length; i++)
-        iFramesElements.push(sortElementsByFocusingIndex(getAllFocusableElements($(sortedIFrames[i]))));
-
-    var elementWithTabIndexFilter = (item, el) => el.tabIndex > 0;
-
-    for (var j = 0; j < elementsArray.length; j++) {
-        results.push(elementsArray[j]);
-
-        if (elementsArray[j].tagName.toLowerCase() === 'iframe') {
-            if (browserUtils.isIE) {
-                results.pop();
-
-                var $iFramesElements     = $(iFramesElements[$.inArray(elementsArray[j], $iFrames)]),
-                    $withTabIndex        = $iFramesElements.filter(elementWithTabIndexFilter),
-                    withTabIndexArray    = $withTabIndex.toArray().sort(sortBy('tabIndex')),
-                    withoutTabIndexArray = $iFramesElements.not($withTabIndex).toArray();
-
-                results = results.concat(withTabIndexArray);
-                results.push(elementsArray[j]);
-                results = results.concat(withoutTabIndexArray);
-            }
-            else {
-                if (browserUtils.isWebKit && iFramesElements[$.inArray(elementsArray[j], $iFrames)].length)
-                    results.pop();
-
-                results = results.concat(iFramesElements[$.inArray(elementsArray[j], $iFrames)]);
-            }
-        }
-    }
-
-    return results;
-}
+export function getNextFocusableElement (element, reverse) {
+    var offset       = reverse ? -1 : 1,
+        allFocusable = sortElementsByFocusingIndex(getFocusableElements(findDocument(element)));
 
-function sortElementsByTabIndex ($elements) {
-    var $withTabIndex = $elements.filter((index, el) => el.tabIndex > 0);
+    //NOTE: in all browsers except Mozilla and Opera focus sets on one radio set from group only.
+    // in Mozilla and Opera focus sets on any radio set.
+    if (element.tagName === "INPUT" && element.type === "radio" && element.name !== "" &&
+        !(browserUtils.isFirefox || browserUtils.isOpera))
+        allFocusable = allFocusable.filter(item => !item.name || item === element || item.name !== element.name);
 
-    if (!$withTabIndex.length)
-        return $elements.toArray();
+    var currentIndex = allFocusable.indexOf(element);
 
-    return $withTabIndex.toArray().sort(sortBy('tabIndex')).concat($elements.not($withTabIndex).toArray());
-}
+    if ((!reverse && currentIndex === allFocusable.length - 1) || (reverse && currentIndex === 0))
```

Comment:
Please simplify expression, by moving it's parts into the variables with the meaningful names